RepoSize incorrect with symlink

Interesting sharness error:

```
make: *** [test/sharness/t0088-repo-stat-symlink.sh] Error 1
make: *** Waiting for unfinished jobs....
*** test/sharness/t0088-repo-stat-symlink.sh ***
# TEST_VERBOSE=1
# TEST_NO_FUSE=1
# TEST_NO_PLUGIN=
# TEST_EXPENSIVE=1
# TEST_OS=LINUX
expecting success: 
  mkdir sym_link_target &&
  ln -s sym_link_target .ipfs

ok 1 - create symbolic link for IPFS_PATH

expecting success: 
    export IPFS_PATH="$(pwd)/.ipfs" &&
    ipfs init --profile=test -b=1024 > /dev/null
  
ok 2 - ipfs init succeeds

expecting success: 
    mkdir mountdir ipfs ipns &&
    test_config_set Mounts.IPFS "$(pwd)/ipfs" &&
    test_config_set Mounts.IPNS "$(pwd)/ipns" ||
    test_fsh cat "\"$IPFS_PATH/config\""
  
ipfs config  "Mounts.IPFS" "/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0088-repo-stat-symlink.sh/ipfs"
ipfs config  "Mounts.IPNS" "/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0088-repo-stat-symlink.sh/ipns"
ok 3 - prepare config -- mounting

expecting success: 
  export IPFS_PATH="sym_link_target" &&
  reposize_direct=$(ipfs repo stat | grep RepoSize | awk '{ print $2 }') &&
  export IPFS_PATH=".ipfs" &&
  reposize_symlink=$(ipfs repo stat | grep RepoSize | awk '{ print $2 }') &&
  test $reposize_symlink -ge $reposize_direct

not ok 4 - 'ipfs repo stat' RepoSize is correct with sym link
#	
#	  export IPFS_PATH="sym_link_target" &&
#	  reposize_direct=$(ipfs repo stat | grep RepoSize | awk '{ print $2 }') &&
#	  export IPFS_PATH=".ipfs" &&
#	  reposize_symlink=$(ipfs repo stat | grep RepoSize | awk '{ print $2 }') &&
#	  test $reposize_symlink -ge $reposize_direct
#	

# failed 1 among 4 test(s)
1..4
test/sharness/Rules.mk:31: recipe for target 'test/sharness/t0088-repo-stat-symlink.sh' failed
```

--- a/test/sharness/lib/test-lib.sh
+++ b/test/sharness/lib/test-lib.sh
@@ -363,8 +363,29 @@ generic_stat() {
     FreeBSD | Darwin | DragonFly)
       _STAT="stat -f %Sp"
       ;;
+    *)
+        echo "unsupported OS" >&2
+        exit 1
+        ;;
   esac
   $_STAT "$1" || echo "failed" # Avoid returning nothing.
+}
+
+# output a file's permission in human readable format
+file_size() {
+    case $(uname -s) in
+        Linux)
+            _STAT="stat --format=%s"
+            ;;
+        FreeBSD | Darwin | DragonFly)
+            _STAT="stat -f%z"
+            ;;
+        *)
+            echo "unsupported OS" >&2
+            exit 1
+            ;;
+    esac
+    $_STAT "$1"
 }
 
 test_check_peerid() {

--- a/test/sharness/t0088-repo-stat-symlink.sh
+++ b/test/sharness/t0088-repo-stat-symlink.sh
@@ -15,14 +15,11 @@ test_expect_success "create symbolic link for IPFS_PATH" '
 
 test_init_ipfs
 
-# compare RepoSize when getting it directly vs via symbolic link
+# ensure that the RepoSize is reasonable when checked via a symlink.
 test_expect_success "'ipfs repo stat' RepoSize is correct with sym link" '
-  export IPFS_PATH="sym_link_target" &&
-  reposize_direct=$(ipfs repo stat | grep RepoSize | awk '\''{ print $2 }'\'') &&
-  export IPFS_PATH=".ipfs" &&
   reposize_symlink=$(ipfs repo stat | grep RepoSize | awk '\''{ print $2 }'\'') &&
-  echo "reposize_symlink: $reposize_symlink;  reposize_direct: $reposize_direct" &&
-  test $reposize_symlink -ge $reposize_direct
+  symlink_size=$(file_size .ipfs) &&
+  test "${reposize_symlink}" -gt "${symlink_size}"
 '
 
 test_done

